src/commonMain/kotlin/fr/acinq/lightning/crypto/LocalKeyManager.kt

```diff
@@ -3,168 +3,173 @@ package fr.acinq.lightning.crypto
 import fr.acinq.bitcoin.*
 import fr.acinq.bitcoin.DeterministicWallet.derivePrivateKey
 import fr.acinq.bitcoin.DeterministicWallet.hardened
+import fr.acinq.bitcoin.crypto.Pack
 import fr.acinq.lightning.Lightning.secureRandom
-import fr.acinq.lightning.channel.ChannelKeys
-import fr.acinq.lightning.channel.RecoveredChannelKeys
-import fr.acinq.lightning.io.Peer
-import fr.acinq.lightning.transactions.Transactions
-
-data class LocalKeyManager(val seed: ByteVector, val chainHash: ByteVector32) : KeyManager {
+import fr.acinq.lightning.NodeParams.Chain
+import fr.acinq.lightning.channel.*
+import fr.acinq.lightning.crypto.LocalKeyManager.Companion.channelKeyPath
+
+/**
+ * An implementation of [KeyManager] that supports deterministic derivation for [KeyManager.ChannelKeys] based
+ * on the initial funding pubkey.
+ *
+ * Specifically, for channel keys there are two paths:
+ *   - `fundingKeyPath`: chosen at random using [newFundingKeyPath]
+ *   - `channelKeyPath`: computed from `fundingKeyPath` using [channelKeyPath]
+ *
+ * The resulting paths looks like so on mainnet:
+ * ```
+ *  node key:
+ *       50' / 0'
+ *
+ *  funding keys:
+ *   - initial funding tx (*):
+ *       50' / 1' / <fundingKeyPath> / <0' or 1'> / 0'
+ *   - splice funding txs (index > 0):
+ *       50' / 1' / <fundingKeyPath> / <0' or 1'> / 6' / <index>
+ *   (*) The initial funding tx is computed differently for historical reasons.
```

Comment:
Same as https://github.com/ACINQ/eclair/pull/2639, I think we should just do:

```suggestion
 *   - initial funding tx:
 *       50' / 1' / <fundingKeyPath> / <0' or 1'> / 0'
 *   - splice funding txs (index > 0):
 *       50' / 1' / <fundingKeyPath> / <0' or 1'> / <index'>
```

Reply:
Agreed. I've updated the doc in 701ea979, but the actual implementation of the funding tx derivation is done in the next PR.